### PR TITLE
Add ServiceRuntime#beforeCommit [ECR-3584]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
## Overview

Add `ServiceRuntime#beforeCommit`.

---
See: https://jira.bf.local/browse/ECR-3584

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [x] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [x] Public API has Javadoc
- [x] Method preconditions are checked and documented in the Javadoc of the method
- [x] Changelog is updated if needed (in case of notable or breaking changes)
- [x] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/core/src/main/java/com/exonum/binding/core/service/Service.java [hunk 1]
```diff
@@ -98,6 +98,21 @@ default void configure(Fork fork, Configuration configuration) {
    */
   void createPublicApiHandlers(Node node, Router router);
 
+  /**
+   * Handles the changes made by all transactions included in the upcoming block.
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/src/main/java/com/exonum/binding/core/service/Service.java]: @alekseysidorov would you check please if the user-facing documentation is accurate?

File: exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/Fork.java [hunk 2]
```diff
@@ -138,13 +150,19 @@ NativeHandle intoPatch() {
 
   /**
    * Creates in-memory checkpoint that can be used to rollback changes.
+   *
+   * <p>Creating a checkpoint will invalidate all collections that were instantiated with this fork.
    */
-  void createCheckpoint() {
+  public void createCheckpoint() {
+    // todo: (1) Are nested rollbacks supported, or is the existing checkpoint lost when a new one
```
Comment (dmitry-timofeev) [inline exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/Fork.java]: ❗️

Comment (vitvakatu) [reply exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/Fork.java]: Nested rollbacks are not possible, only one checkpoint is available at time.

Comment (dmitry-timofeev) [reply exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/Fork.java]: OK, I submitted a [task](https://jira.bf.local/browse/ECR-3611) to consider to hide this operation, but I had initially misunderstood how it works and the possible consequences: as multiple rollbacks do not reset the state to the initial fork state, changes made by other services won't be affected.

Comment (dmitry-timofeev) [reply exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/Fork.java]: The new test and the updated documentation must hopefully clarify the things.

**Pull request merged**